portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -1,25 +1,87 @@
 package com.google.sps.servlets;
 
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.util.Date;
 import java.io.IOException;
+import java.io.InputStream;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
 
 @WebServlet("/form-handler")
+@MultipartConfig
 public class FormHandlerServlet extends HttpServlet {
 
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response)throws ServletException, IOException {
     String name = request.getParameter("name-input");
     String email = request.getParameter("email-input");
     String message = request.getParameter("message-input");
 
+    // Get date from server
+    Date date = new Date();
+    String dateTime = date.toString();
+
+     // Get the file chosen by the user.
+     Part filePart = request.getPart("image");
+     String uploadedFileUrl = "—";
+ 
+     //Check if there is a file uploaded
+     if(filePart.getSize()>0){
+        String fileName = filePart.getSubmittedFileName();
+        InputStream fileInputStream = filePart.getInputStream();
+ 
+        // Upload the file and get its URL
+        uploadedFileUrl = uploadToCloudStorage(fileName, fileInputStream);
+     }
+
     System.out.println("name: " + name);
     System.out.println("email: " + email);
     System.out.println("message: " + message);
 
+    //  Writing to Datastore
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("ContactInfo");
+    FullEntity taskEntity = 
+                            Entity.newBuilder(keyFactory.newKey())
+                            .set("Name", name)
+                            .set("Email", email)
+                            .set("Message", message)
+                            .set("File", uploadedFileUrl)
+                            .set("Date", dateTime)
+                            .build();
+    datastore.put(taskEntity);
+
+    // Redirecting the website 
     response.sendRedirect("https://swu-sps-summer22.appspot.com/#contact");
```

Comment:
I think in this case it is better to redirect to relative path: sendRedirect("/#contact");
This way you can deploy the same implementation to multiple domains without changing your server implementation.